public_html/question2.js

```diff
@@ -0,0 +1,41 @@
+let n = prompt("Enter a number (n*n grid)");
```

Comment:
One of the goals for this assignment was not to use global variables. Declaring this variable outside of any scope is declaring it as a global scope. The one you did in the question one by enclosing it inside IIFE, is a way to create an anonymous scope, and variables declared inside it will not be in the global scope.